### PR TITLE
task-driver: redeem-realyer-fee: Define task skeleton

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR implements the skeleton of the `redeem-relayer-fee` task. This task will be used by relayers to settle notes from chain state into their own wallets.

Implementation is deferred to a follow-up

### Testing
- All unit tests pass

### CODE REVIEW

File: common/src/types/tasks.rs [hunk 1]
```diff
@@ -422,6 +428,31 @@ impl From<PayRelayerFeeTaskDescriptor> for TaskDescriptor {
...
 }
 
+/// The task descriptor for redeeming a relayer note
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct RedeemRelayerFeeTaskDescriptor {
+    /// The wallet ID of the relayer's wallet
+    ///
+    /// Technically this should be static and not needed here, but we include it
+    /// to allow the descriptor struct to compute its own task queue key
+    pub wallet_id: WalletIdentifier,
```
Comment (akirillo) [inline common/src/types/tasks.rs:438]: why can't we use a `static` in computing the task queue key?

File: workers/task-driver/src/tasks/redeem_relayer_fee.rs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+/// Defines the redeem relayer fee task
+pub struct RedeemRelayerFeeTask {
+    /// The note to redeem
+    pub note: Note,
+    /// The proof of `VALID FEE REDEMPTION` used to pay the fee
+    pub proof: Option<FeeRedemptionBundle>,
+    /// The arbitrum client used for submitting transactions
+    pub arbitrum_client: ArbitrumClient,
+    /// A hand to the global state
```
Comment (akirillo) [inline workers/task-driver/src/tasks/redeem_relayer_fee.rs:119]:
```suggestion
    /// A handle to the global state
```

**Review by akirillo — APPROVED** [review]

File: common/src/types/tasks.rs (hunk 1 above)
Comment (joeykraut) [reply common/src/types/tasks.rs:438]: this value isn't static, it may be generated on startup

Comment (akirillo) [reply common/src/types/tasks.rs:438]: ah, blanked out - we'd have to fetch this from state, right, which is async and doesn't play well with the `get_task_queue_key` interface?

Comment (akirillo) [reply common/src/types/tasks.rs:438]: i suppose we could put it in a `OnceCell` on startup, but eh, inconsequential

**Pull request merged**